src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -13,10 +13,24 @@ namespace System.Net.NetworkInformation
     public class NetworkChange
     {
         private static NetworkAddressChangedEventHandler s_addressChangedSubscribers;
+        private static NetworkAvailabilityChangedEventHandler s_availabilityChangedSubscribers;
         private static volatile int s_socket = 0;
         private static readonly object s_lockObj = new object();
+        private static readonly object s_availabilityLock = new object();
```

Comment:
The new lock was basically just used to control access to the "availability has changed" flag. I'll take a look at this again tomorrow and rename / comment things more appropriately.